csrc/python_frontend/python_bindings.cpp

```diff
@@ -2665,7 +2665,8 @@ void initNvFuserPythonBindings(PyObject* module) {
       "cat",
       [](FusionDefinition::Operators& self,
          std::vector<Tensor> tensors,
-         int64_t dim) -> Tensor {
+         int64_t dim,
+         bool manual_padding) -> Tensor {
```

Comment:
Do we need this? Shouldn't python API to always have the vanilla `cat` that lowers to `pad + cat`?

Follow up with Naoya's question
> How to know if a certain subset of ops corresponds to a cat?

I think the case we are making here is that, we assume that `cat` is always lowered to `PadOp + CatOp`. So any `PadOp` followed by `CatOp` is considered as `corresponds to a cat`.

We just needed a different API to skip adding the `PadOp`, since when we do cpp->python translation, we've already explicitly added the PadOp in the fusion.

Reply:
The translated python definition will have a CatOpRecord that needs print a python string with `manual_padding` set to true.